sql/sql_insert.cc

```diff
@@ -4317,7 +4317,17 @@ bool select_insert::prepare_eof()
   if (info.ignore || info.handle_duplicates != DUP_ERROR)
       if (table->file->ha_table_flags() & HA_DUPLICATE_POS)
         table->file->ha_rnd_end();
-  table->file->extra(HA_EXTRA_END_ALTER_COPY);
+  int create_err= table->file->extra(HA_EXTRA_END_ALTER_COPY);
+  if (error <= 0)
```

Comment:
Why is the condition not `if (!error)`? Some overridden `handler:end_bulk_insert()` could return a negative error code.

Reply:
error is set to -1 when handler encounters HA_ERR_END_OF_FILE. So error variable is > 0 then we encounter error.